### ISSUE TITLE
Embedding 模型 Lora Merge 前后推理结果不一致

### ISSUE THREAD

**Issue by liyi01827** [modelscope/ms-swift#5703]
推理代码
`from swift.llm import (
    PtEngine, RequestConfig, safe_snapshot_download, get_model_tokenizer, get_template, InferRequest
)
from swift.tuners import Swift
from swift.utils import copy_files_by_pattern

import copy

from transformers import AutoProcessor, AutoModelForCausalLM
processor = AutoProcessor.from_pretrained("Qwen/Qwen2.5-VL-3B-Instruct")

prompt = "请描述这两张图片"
conversation = [
        {
            "role": "user",
            "content": [
                {"type": "image", "image": "./dog.jpg"},
                {"type": "image", "image": "./dog.jpg"},
                {"type": "text", "text": prompt},
            ],
        }
    ]

a = processor.tokenizer.encode(processor.apply_chat_template(conversation, add_generation_prompt=True))

model_dir = './Qwen25-VL-3B-Instruct'
lora_checkpoint_dir = './checkpoint-3670'  # 修改成checkpoint_dir

model1, tokenizer = get_model_tokenizer(model_dir, model_type='qwen2_5_vl')
model1.eval()
model1 = Swift.from_pretrained(model1, lora_checkpoint_dir)
model1.eval()

model2, _ = get_model_tokenizer('./checkpoint-3670-merged', model_type='qwen2_5_vl')
model2.eval()

import torch
with torch.no_grad():
    o1 = model1(torch.tensor(a).view(1, -1), temperature=0)
    o2 = model2(torch.tensor(a).view(1, -1), temperature=0)

print(o1['logits'][0,-1,:])
print(o2['logits'][0,-1,:])
`

推理的embedding输出结果
`tensor([ 2.0000, -2.6562,  0.6523,  ...,  1.6562,  0.8398, -3.7031],
       dtype=torch.bfloat16)

tensor([ 1.9688, -2.7188,  0.6445,  ...,  1.6094,  0.7734, -3.7812],
       dtype=torch.bfloat16)`

这里base模型和lora都是bf16训练的。请问下出现这个问题可能的原因是什么？ 哪个地方可能有bug吗？ 版本是3.8.0.dev0

**Comment by FlyingYanglu** [comment]
^

**Comment by ve9etandy** [comment]
Same issue

**Comment by tastelikefeet** [comment]
embedding结果看起来比较相似，应该是merge过程中的问题。lora ckpt的dtype可能是float32，如果原模型是float16，合并后会有精度损失，可否检查下lora ckpt的类型来确认下

The embedding results seem similar, it should be a problem of merging.
I think the dtype of lora adapters should be float32, can you check the safetensor files to get the dtype?
If the dtype of lora ckpts is float32, and the original model is float16/bf16, then there will be a loss of precision.

**Comment by z0o0ey** [comment]
please retry with latest swift and vllm, feel free to reopen or create a new issue if it the problem persists.
